src/components/cesium_map/cesium_UI.js

```diff
@@ -340,17 +342,22 @@ class CesiumMap extends React.Component {
     // set time interval to check the current view every 5 seconds and update points
     this.checkPosition = setInterval(() => {
       const loading = document.getElementById("loading").style.display;
-      let diffDistance = distanceInKm(
+      const diffDistanceMove = distanceInKm(
         cameraLat,
         cameraLong,
         viewer.currentView.latitude,
         viewer.currentView.longitude);
-      // update the points every 5 seconds if two points differ in 50km + scale of height.
-      // I scale the current height by 15000000, the height of "View Global".
-      // 4000 km is the distance that rotate half earth map on the height 15000000.
+      const diffDistanceFarthest = distanceInKm(
+        setPrimitive.farthest.y,
+        setPrimitive.farthest.x,
+        cameraLat,
+        cameraLong);
+      // update the points every 5 seconds
       // Update:
       //      A new parameter loading to indicate if the users cick somewhere and avoid intervel to check positions.
-      if (loading && diffDistance > MINIMUM_REFRESH_DISTANCE + MAXIMUM_REFRESH_DISTANCE * viewer.currentView.height / MAXIMUM_ZOOM_DISTANCE) {
+      // New method:
+      //      The update condition is based on if the move distance is larger than the ratio of radius of the primitive points.
+      if (loading && diffDistanceMove > diffDistanceFarthest * UPDATE_RATIO) {
```

Comment:
New method: the Cesium map would only update the primitive points when the camera move distance is larger than the ratio of the radius of the primitive point collection. Our API would return the point positions based on the Geo distances so the points would distribute as a circle. `PointStreamPrimitiveCollection` class has a new `get` function to return the last point position. `diffDistanceMove` is the distance from the original camera position to the camera move position. `diffDistanceFarthest` is the distance from the original camera position to the farthest point which is the largest radius. The old method would update points unexpectedly. The new method is more user-friendly.

Reply:
This makes sense - I'll try it out locally